plugin/src/com/microsoft/alm/plugin/external/visualstudio/VisualStudioTfvcClient.java

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root.
+
+package com.microsoft.alm.plugin.external.visualstudio;
+
+import com.intellij.openapi.diagnostic.Logger;
+import com.intellij.openapi.util.SystemInfo;
+import com.intellij.openapi.util.text.StringUtil;
+import com.microsoft.alm.plugin.external.exceptions.VisualStudioClientVersionException;
+import com.microsoft.alm.plugin.external.models.ToolVersion;
+import com.microsoft.alm.plugin.services.PropertyService;
+import org.apache.commons.lang.StringUtils;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CompletionStage;
+
+public class VisualStudioTfvcClient {
+
+    private static final Logger ourLogger = Logger.getInstance(VisualStudioTfvcClient.class);
+
+    private static final String[] SUPPORTED_VS_VERSIONS = new String[] { "2019", "2017" };
+    private static final String[] SUPPORTED_VS_EDITIONS = new String[] { "Enterprise", "Professional", "Community" };
+    private static final String EDITION_RELATIVE_TF_EXE_PATH =
+            "Common7\\IDE\\CommonExtensions\\Microsoft\\TeamFoundation\\Team Explorer\\TF.exe";
+
+    private static final ToolVersion MINIMAL_SUPPORTED_VERSION = new ToolVersion(15, 0, 0, "");
```

Comment:
It looks little strange. You use version constant here and 2019 2017 five lines above. But 15 directly connected with Visual Studio version maybe we should handle it?

Reply:
Maybe we should point ti the TF Client version.

Reply:
Yes, version 15 is VS 2017 (I didn't bother to check older releases, so let's warn they're unsupported — essentially, we didn't check if they work well).

I agree that we could state more clearly that we only officially support client from VS2017 onwards. I'll improve that.